propagation/http/http.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2019 The OpenZipkin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"strings"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/openzipkin/zipkin-go/model"
+	zb3 "github.com/openzipkin/zipkin-go/propagation/b3"
+)
+
+const (
+	traceIDHeader      = "x-b3-traceid"
+	spanIDHeader       = "x-b3-spanid"
+	parentSpanIDHeader = "x-b3-parentspanid"
+	sampledHeader      = "x-b3-sampled"
+	flagsHeader        = "x-b3-flags"
+)
+
+// Propagator exposes default B3 http propagator implementation
+var Propagator = &propagator{}
+
+type propagator struct{}
+
+func (p *propagator) Inject(sc model.SpanContext, carrier interface{}) error {
+	if (model.SpanContext{}) == sc {
```

Comment:
zipkin-go doesn't have an extraction utility?

Reply:
yes but signature is different so we needed to replicate some of it here.